### PR TITLE
Fix some shellcheck warnings/errors in cluster/gce/util.sh

### PR DESCRIPTION
**Pull request by joakimr-axis**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes shellcheck failures in cluster/gce/util.sh

**Which issue(s) this PR fixes**:
ref: #72956 (part of #90405)
Clean up the use of arithmetic shell functionality.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -2066,20 +2066,16 @@ function create-node-template() {
...
       validate-node-local-ssds-ext "${ssdopts}"
-      for i in $(seq ${ssdopts[0]}); do
+      for ((i=1; i<=ssdopts[0]; i++)); do
         local_ssds="$local_ssds--local-ssd=interface=${ssdopts[1]} "
       done
     done
   fi
 
-  if [[ ! -z ${NODE_LOCAL_SSDS+x} ]]; then
-    # The NODE_LOCAL_SSDS check below fixes issue #49171
```
Comment (mm4tt) [inline cluster/gce/util.sh:2076]: Can we keep the comment so someone doesn't re-introduce the bug when refactoring?

**Review by mm4tt — COMMENTED** [review]
/ok-to-test

File: cluster/gce/util.sh (hunk 1 above)
Comment (joakimr-axis) [reply cluster/gce/util.sh:2076]: Yes, I'll fix it immediately!

**Pull request merged**